File: OG/P2_A_Identifier.java
import java.util.*;

class ident {
    static boolean isValid(String str, int n) {

        // If first character is invalid
        if (!((str.charAt(0) >= 'a' && str.charAt(0) <= 'z')
                || (str.charAt(0) >= 'A' && str.charAt(0) <= 'Z')))
            return false;

        // Traverse the string for the rest of the characters
        for (int i = 1; i < str.length(); i++) {
            if (!((str.charAt(i) >= 'a' && str.charAt(i) <= 'z')
                    || (str.charAt(i) >= 'A' && str.charAt(i) <= 'Z')
                    || (str.charAt(i) >= '0' && str.charAt(i) <= '9')
                    || str.charAt(i) == '_'))
                return false;
        }

        // String is a valid identifier
        return true;
    }

    public static void main(String args[]) {
        System.out.println("Saail Chavan KFPMSCCS016\n");
        System.out.println("Enter a string");
        Scanner c = new Scanner(System.in);
        String str = c.nextLine();
        int n = str.length();

        if (isValid(str, n))
            System.out.println("Identifier");
        else
            System.out.println("Not an Identifier");
    }
}

File: Py/P2_A_Identifier.py
#Design a program to check given input is identifier, constants, reserved keywords, and operators.
#Identifier

def is_valid_identifier(s):
    # Check if the first character is a letter
    if not (('a' <= s[0] <= 'z') or ('A' <= s[0] <= 'Z')):
        return False

    # Traverse the string for the rest of the characters
    for i in range(1, len(s)):
        if not (('a' <= s[i] <= 'z') or ('A' <= s[i] <= 'Z') or ('0' <= s[i] <= '9') or s[i] == '_'):
            return False

    # String is a valid identifier
    return True


if __name__ == "__main__":
    print("Saail Chavan KFPMSCCS016")
    str_input = input("Enter a string: ")

    if is_valid_identifier(str_input):
        print("Identifier")
    else:
        print("Not an Identifier")
